### PR TITLE
Add webhookname from env

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

Ref: https://github.com/knative/eventing/issues/4530

This is currently used here
https://github.com/knative/eventing/blob/52597d174b52a680756b52b377c81bfb96fb636b/pkg/logconfig/config.go#L37

Moving it to pkg to be used across all knative components

### CODE REVIEW

File: webhook/env.go [hunk 1]
```diff
@@ -37,3 +40,17 @@ func PortFromEnv(defaultPort int) int {
 	}
 	return port
 }
+
+func NameFromEnv() string {
+	if webhook := os.Getenv(webhookNameEnv); webhook != "" {
+		return webhook
+	}
+
+	panic(fmt.Sprintf(`The environment variable %q is not set
```
Comment (vagababov) [inline webhook/env.go:49]: Why would this be required? Can't it be sensibly defaulted?

Comment (arghya88) [reply webhook/env.go:49]: As it's being done for the `PortFromEnv` ? or you mean just set a hardcoded default value if anything is not set in env?

Comment (mattmoor) [reply webhook/env.go:49]: @vagababov I think requiring it is better than defaulting it.  If we default it for two components in a namespace, then they will end up with the same lease names and have a bad time.  I helped the Tekton folks debug this with pipelines and triggers collocated recently.

We should probably change the panic message to indicate that this should be unique for the webhooks in a namespace.

File: webhook/env.go [hunk 2]
```diff
@@ -37,3 +40,17 @@ func PortFromEnv(defaultPort int) int {
...
+If this is a process running on Kubernetes, then it should be using the downward
+API to initialize this variable via:
```
Comment (mattmoor) [inline webhook/env.go:51]: This isn't the downward API 😂 

I think this is carryover from `system.Namespace()` 🤦

Comment (arghya88) [reply webhook/env.go]: Done some changes.PTAL

**Review by mattmoor — APPROVED** [review]
/lgtm

**Pull request merged**